Solutions/STM32F4DISCOVERY/scatterfile_gcc_missing_symbols.xml

```diff
@@ -1,5 +1,10 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <ScatterFile xmlns="http://schemas.microsoft.com/netmf/ScatterfileSchema.xsd">
+  <!-- 
+    - using newlib-nano adds libnosys.a, which calls sbrk, which requires the 'end' symbol
+    - this is pulled in by a _printf imported in function _Crypto_StepRSAOperation @ Crypto.lib(rsa_zenith.obj) 
+  -->
+  <Provide Name="end" />

```

Comment:
Ha, no it's not done work. Just an investigation to see about using mbedTLS to create an OSS implementation of the crypto lib support. 

As to how many devices use the crypto support I don't know. If they use the firmware image signing and TinyBooter that requires the cryptolibs. I believe, but haven't yet verified that the MFUpdate mechanism doesn't use the crypto libs signature/hashing so could be replaced. At this point I'm leaning towards just pulling the old crypto support as there's a lot in the mix for the next round of NETMF to warrant a new major label (for example v5.0.0 ) where a lot of legacy cruft can be ejected. But much of that depends on actual users to provide feedback on what is used/needed.